### PR TITLE
Add tolerations and affinity on fluentd Daemonset

### PR DESCRIPTION
**Pull request by kaxil**
Based on the change in https://github.com/astronomer/helm.astronomer.io/pull/241

We need to wait on merging this PR until a new release/tag of our helm chart is released !

### CODE REVIEW

**Comment by sjmiller609** [comment]
I don't think this change is necessary because fluentd is already restricted to nodes with multi-tenant=true

https://github.com/astronomer/terraform-google-astronomer-cloud/blob/da64c2fd17c56bfdd439d5d934246e95e94ba558/locals.tf#L26-L35

**Comment by sjmiller609** [comment]
ah ok. looking at the other PR now

File: locals.tf [hunk 1]
```diff
@@ -229,6 +229,29 @@ prometheus:
...
+              - key: "astronomer.io/multi-tenant"
+                operator: In
+                values:
+                  - "true"
+%{if var.create_dynamic_pods_nodepool == true}
+          - matchExpressions:
+              - key: "astronomer.io/dynamic-pods"
+                operator: In
+                values:
+                  - "true"
```
Comment (sjmiller609) [inline locals.tf]: I think we can delete these lines, since we opted to include multi-tenant label in the new node pool

File: locals.tf [hunk 2]
```diff
@@ -229,6 +229,29 @@ prometheus:
     limits:
       cpu: "2000m"
       memory: "16Gi"
+fluentd:
+  nodeSelector: {}
```
Comment (sjmiller609) [inline locals.tf]: can we delete this line?

File: locals.tf (hunk 1 above)
Comment (kaxil) [inline locals.tf]: This is no longer needed I suppose as we have not removed the multi-tenant label from the New NodePool in https://github.com/astronomer/terraform-google-astronomer-gcp/pull/34

Comment (sjmiller609) [reply locals.tf]: sounds like you beat me to it Kaxil!

**Comment by sjmiller609** [comment]
seems related https://github.com/terraform-providers/terraform-provider-google-beta/releases

**Comment by kaxil** [comment]
Updated.

I am not sure of the CI error, maybe just network connectivity!:

```

Initializing provider plugins...
--
15 | - Checking for available provider plugins...
16 | - Downloading plugin for provider "acme" (terraform-providers/acme) 1.5.0...
17 | - Downloading plugin for provider "null" (hashicorp/null) 2.1.2...
18 | - Downloading plugin for provider "http" (hashicorp/http) 1.1.1...
19 | - Downloading plugin for provider "random" (hashicorp/random) 2.2.1...
20 | - Downloading plugin for provider "google" (hashicorp/google) 3.0.0-beta.1...
21 |  
22 | Error installing provider "google": bad response code: 403.
23 |  
24 | Terraform analyses the configuration and state and automatically downloads
25 | plugins for the providers used. However, when attempting to download this
26 | plugin an unexpected error occurred.
27 |  
28 | This may be caused if for some reason Terraform is unable to reach the
29 | plugin repository. The repository may be unreachable if access is blocked
30 | by a firewall.
31 |  
32 | If automatic installation is not possible or desirable in your environment,
33 | you may alternatively manually install plugins by downloading a suitable
34 | distribution package and placing the plugin's executable file in the
35 | following directory:
36 | terraform.d/plugins/linux_amd64
37 |  
38 | - Downloading plugin for provider "kubernetes" (hashicorp/kubernetes) 1.10.0...
39 | - Downloading plugin for provider "tls" (hashicorp/tls) 2.1.1...
40 | - Downloading plugin for provider "helm" (hashicorp/helm) 0.10.4...
41 | - Downloading plugin for provider "google-beta" (terraform-providers/google-beta) 3.0.0-beta.1...
42 | - Downloading plugin for provider "local" (hashicorp/local) 1.4.0...
43 |  
44 | Error: bad response code: 403


```

**Comment by sjmiller609** [comment]
I think because they just published a new version

**Comment by kaxil** [comment]
> seems related https://github.com/terraform-providers/terraform-provider-google-beta/releases

Yeah !

**Comment by kaxil** [comment]
Passes now @sjmiller609

**Comment by schnie** [comment]
The rest of the PRs have been merged but not released yet.

**Comment by sjmiller609** [comment]
I version locked the google providers in the CI of *-gcp and this repo to 2.20.0 and updated in google-environoments the existing lock from ~> 2.16 to ~> 2.20

**Comment by schnie** [comment]
Yep, trying to fix the airflow build and will be cutting a release.

**Comment by schnie** [comment]
@kaxil we want to remove `deploymentNodePool` from our config now that we shifted it down to `fluentd` right? I don't think its being used anymore.

**Comment by kaxil** [comment]
> @kaxil we want to remove `deploymentNodePool` from our config now that we shifted it down to `fluentd` right? I don't think its being used anymore.

Updated

**Comment by kaxil** [comment]
Rebased on the master @schnie @sjmiller609

**Comment by sjmiller609** [comment]
Thanks kaxil

**Comment by sjmiller609** [comment]
this PR requires this other PR to be merged first https://github.com/astronomer/helm.astronomer.io/pull/298

**Pull request merged**